mosdef_cassandra/core/moves.py

```diff
@@ -173,6 +181,94 @@ def __init__(self, ensemble, species_topologies):
                 self.prob_translate += self.prob_rotate
                 self.prob_rotate = 0.0
 
+    @property
+    def restricted_type(self):
+        return self._restricted_type
+
+    @restricted_type.setter
+    def restricted_type(self, restricted_type):
+        if restricted_type:
+            if self.ensemble == 'gcmc':
+                if len(restricted_type) != 1:
+                    raise ValueError("{} ensemble only has 1 box"
+                    ' but restricted_insertion of length {}'
+                    ' was passed.'.format(
+                        self.ensemble,
+                        len(restricted_type)))
+                if not isinstance(restricted_type[0], (list, tuple)):
+                    raise TypeError("'restricted_type' specified for"
+                    " a box must be an iterable.")
+                if len(restricted_type[0]) != self._n_species:
+                    raise ValueError("{} species specified"
+                    ' but restricted_type for {}'
+                    ' species was passed.'.format(
+                        self._n_species,
+                        len(restricted_type[0])))
+            elif self.ensemble in ['gemc', 'gemc_npt']:
+                if len(restricted_type) != 2:
+                    raise ValueError("{} ensemble requires 2 boxes"
+                    ' but restricted_type of length {}'
+                    ' was passed.'.format(
+                        self.ensemble,
+                        len(restricted_type)))
+                if len(restricted_type[0]) != self._n_species:
+                    raise ValueError("{} species specified"
+                    ' but restricted_typed for only {}'
+                    ' species was passed.'.format(
+                        self._n_species,
+                        len(restricted_type[0])))
+            else:
+                raise ValueError("Restricted insertions are only valid"
+                    ' for GCMC, GEMC, and GEMC-NPT ensembles.'
+                    ' The {} ensemble was passed.'.format(
+                        self.ensemble))
+
+            self._restricted_type = restricted_type
+        else:
+            self._restricted_type = restricted_type
+
+    @property
+    def restricted_value(self):
+        return self._restricted_value
+
+    @restricted_value.setter
+    def restricted_value(self, restricted_value):
+        if restricted_value:
+            if len(self.restricted_type) != len(restricted_value):
+                raise ValueError("'restricted_type' is a list of length"
+                        " {} but 'restricted_value' is a list of length"
```

Comment:
Logically the comparison makes sense, but I think the error message could be a bit clearer. The underlying problem is that the user is specifying restricted values for the wrong number of boxes.